scripts/pkg/build_templates/opensearch/deb/debian/preinst

```diff
@@ -23,6 +25,28 @@ if command -v systemctl >/dev/null && systemctl is-active opensearch-performance
     systemctl --no-reload stop opensearch-performance-analyzer.service
 fi
 
+# Check if OPENSEARCH_INITIAL_ADMIN_PASSWORD is defined
+# TODO:
+# 1. This check will need to be modified if there will be a min dist for deb in future (currently there is none)
+# 2. Currently, the demo config setup is defined to run, in postinst, if `opensearch-security` is present. Cannot apply the same check here since the plugins folder is not available yet.
+
+# Check if this is an upgrade by checking whether opensearch already exists
+set +e
+dpkg-query -l opensearch | grep ^ii
+OPENSEARCH_INSTALLED=$? # 0 when installed, 1 when installed
+set -e
+
+OPENSEARCH_REQUIRED_VERSION="2.12.0"
+MINIMUM_OF_TWO_VERSIONS=`echo $OPENSEARCH_REQUIRED_VERSION $OPENSEARCH_VERSION | tr ' ' '\n' | sort -V | uniq | head -n 1`
+
+if [ $OPENSEARCH_INSTALLED = 1 ] && [ $OPENSEARCH_VERSION != CHANGE_VERSION ]; then
```

Comment:
The issue I mentioned is not addressed: `$OPENSEARCH_VERSION` is set to `CHANGE_VERSION` on line 14, both occurrences of `CHANGE_VERSION` are replaced by an actual version, so in the end `$OPENSEARCH_VERSION != CHANGE_VERSION` is always false.

Reply:
I added this check as "just to be sure" but I'll remove it the CHANGE_VERSION will always be replaced with actual version value.